src/decode.rs

```diff
@@ -4587,172 +4587,54 @@ unsafe fn decode_b(
-                                < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.matrix[5 as libc::c_int as usize]),
-                            if (t.warpmv.u.p.alpha as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.alpha as libc::c_int),
-                            if (t.warpmv.u.p.beta as libc::c_int) < 0 as libc::c_int {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.beta as libc::c_int),
-                            if (t.warpmv.u.p.gamma as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.gamma as libc::c_int),
-                            if (t.warpmv.u.p.delta as libc::c_int) < 0 as libc::c_int
-                            {
-                                '-' as i32
-                            } else {
-                                ' ' as i32
-                            },
-                            abs(t.warpmv.u.p.delta as libc::c_int),
-                            b
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .mv2d
-                                .c2rust_unnamed
-                                .y as libc::c_int,
-                            b
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .mv2d
-                                .c2rust_unnamed
-                                .x as libc::c_int,
+                        use std::fmt;
+
+                        /// Helper struct for printing a number as a signed hexidecimal value.
+                        struct SignAbs(i32);
+
+                        impl fmt::Display for SignAbs {
+                            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                                let sign = if self.0 < 0 { "-" } else { " " };
+                                write!(f, "{}{:x}", sign, self.0.abs())
+                            }
+                        }
```

Comment:
`signabs` was used twice in `decode_b`, though re-defined.  I think we can move this to the top of `decode_b`.